### PR TITLE
feat: origin

### PR DESCRIPTION
**Pull request by szmarczak**

### CODE REVIEW

File: lib/fetch/util.js [hunk 1]
```diff
@@ -227,9 +227,109 @@ function appendFetchMetadata (httpRequest) {
   //  TODO
 }
 
+// https://url.spec.whatwg.org/#concept-host-serializer
+function serializeHost (host) {
+  // 1. If host is an IPv4 address, return the result of running the IPv4 serializer on host.
```
Comment (szmarczak) [inline lib/fetch/util.js:232]: We store IPs as strings, no?

Comment (szmarczak) [reply lib/fetch/util.js]: Ah, WHATWG URL is already taking care of origin normalization :)

**Comment by ronag** [comment]
LGTM, any reason still it's still draft?

**Comment by ronag** [comment]
thx!

**Pull request merged**